pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -96,37 +96,50 @@ func machinesEqual(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
 	return m1.Name == m2.Name && m1.Namespace == m2.Namespace
 }
 
+// isNodeJoin determines if a machine, in scope, should join of the cluster.
 func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachines []*clusterv1.Machine) (bool, error) {
 	switch set := scope.Machine.ObjectMeta.Labels["set"]; set {
```

Comment:
Not directly related to this PR, but instead of switching on the `set` label, this could just check if `scope.Machine.Spec.Versions.ControlPlane != ""`

Reply:
The idea is to determine whether or not this machine in scope should join the cluster or not and also be able to handle all "kinds" of machines- controlplane, worker, and any other we might add later. 

`scope.Machine.Spec.Versions.ControlPlane != ""` will just tell us whether this is a controlplane machine or not.

Reply:
Yes, I do think that we should be able to potentially special case other solutions as well, but currently we have two different ways of determining a control-plane machine within cluster-api upstream (for the purposes of clusterctl) and per-provider. If we can standardize on a single solution for controlplane or not, then it is less cognitive overhead for the end user. 

Removing the requirement of the `set` label for control plane instances would also reduce user confusion when trying to deploy a MachineSet or MachineDeployment, where it isn't overly clear that they need to include those in the labels for the template.

Reply:
I agree on not having 2 ways of doing the same. As it is not directly related to this PR, I'll follow-up with a PR for this.